tests/pytorch/test_dataloader.py

```diff
@@ -130,47 +128,44 @@ def _check_device(data):
 
 @parametrize_idtype
 @pytest.mark.parametrize('sampler_name', ['full', 'neighbor', 'neighbor2'])
-@pytest.mark.parametrize('pin_graph', [None, 'cuda_indices', 'cpu_indices'])
-def test_node_dataloader(idtype, sampler_name, pin_graph):
+@pytest.mark.parametrize('mode', ['cpu', 'uva_cuda_indices', 'uva_cpu_indices', 'pure_gpu'])
+@pytest.mark.parametrize('use_ddp', [False, True])
```

Comment:
Perhaps we should break this test into multiple tests to simplify the logic and make the code more readable.

Reply:
All these options need to be tested with the GPU backend, so I don't expect it would be simpler by splitting CPU and GPU tests.